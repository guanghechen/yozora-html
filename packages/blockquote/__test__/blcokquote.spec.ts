import type { Blockquote, Emphasis, Strong, Text } from '@yozora/ast'
import { BlockquoteType, EmphasisType, StrongType, TextType } from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderBlockquote from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node: Blockquote = {
      type: BlockquoteType,
      children: [
        {
          type: TextType,
          value: 'yozora is cool!',
        } as Text,
      ],
    }
    expect(renderBlockquote(node, renderChildren)).toMatchSnapshot()
  })

  it('basic-2', function () {
    const node: Blockquote = {
      type: BlockquoteType,
      children: [
        {
          type: EmphasisType,
          children: [
            {
              type: TextType,
              value: 'Hello',
            } as Text,
          ],
        } as Emphasis,
        {
          type: TextType,
          value: ', ',
        } as Text,
        {
          type: StrongType,
          children: [
            {
              type: TextType,
              value: 'World',
            } as Text,
          ],
        } as Strong,
      ],
    }
    expect(renderBlockquote(node, renderChildren)).toMatchSnapshot()
  })
})
