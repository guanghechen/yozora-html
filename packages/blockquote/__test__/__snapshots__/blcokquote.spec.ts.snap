// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<blockquote class="yozora-blockquote"><span class="yozora-text">yozora is cool!</span></blockquote>"`;

exports[`snapshot basic-2 1`] = `"<blockquote class="yozora-blockquote"><em class="yozora-emphasis"><span class="yozora-text">Hello</span></em><span class="yozora-text">, </span><strong class="yozora-strong"><span class="yozora-text">World</span></strong></blockquote>"`;
